Library/Homebrew/brew.rb

```diff
@@ -32,10 +32,12 @@ class MissingEnvironmentVariables < RuntimeError; end
   raise e if ENV["HOMEBREW_MISSING_ENV_RETRY"]
 
   if ENV["HOMEBREW_DEVELOPER"]
-    $stderr.puts <<~EOS
-      Warning: #{e.message}
-      Retrying with `exec #{ENV["HOMEBREW_BREW_FILE"]}`!
-    EOS
+    Tty.with($stderr) do |stderr|
```

Comment:
I think it's only worth doing this for when we're outputting colours etc.

Reply:
@reitermarkus suggested that we use `Tty.with` for all the `puts e` statements, as some of them have the use of `Formatter` in their `#to_s` methods. It might be safe to remove thus behaviour for `e.message` and `e.backtrace`, though.

Reply:
> It might be safe to remove this behaviour for `e.message` and `e.backtrace`, though.

Yeh, I think so 👍🏻. Let's assume so for now.

Reply:
Isn't `e.message` the same as `e.to_s` (or may be)? But yeah, `e.backtrace` definitely doesn't need it.

Reply:
> Isn't `e.message` the same as `e.to_s` (or may be)?

Yeh, fair point 👍🏻 



> But yeah, `e.backtrace` definitely doesn't need it.

Agreed 👍🏻